### ISSUE TITLE
Error compiling fragment shader

### ISSUE THREAD

**Issue by mad1ost** [gpujs/gpu.js#421]
I include `gpu-browser.js` in my index.html page like this
```
<script src='/path/to/js/gpu-browser.js'></script>
```
and I try to run the following code from Documentation in WebConsole of my browser (Chrome or Firefox 65), but get the following error
```
const gpu = new GPU();

const myFunc = gpu.createKernel(function() {
     const array2 = [0.08, 2];
     return array2;
 }).setOutput([100]);

myFunc();
gpu-browser.js:9002 Uncaught Error: Error compiling fragment shader: ERROR: 0:143: '=' : dimension mismatch
ERROR: 0:143: 'assign' : cannot convert from 'highp 2-component vector of float' to 'highp float'
```

**Comment by robertleeplummerjr** [comment]
Looking at this now.

**Comment by robertleeplummerjr** [comment]
This has to do specifically with: https://github.com/gpujs/gpu.js/issues/397
In short, it isn't support _yet_, however, I believe this will work, although not very fun:

```js
const myFunc = gpu.createKernelMap([function lastIndex(value) { return value; }], function() {
  const array2 = [0.08, 2];
  const firstIndex = array2[0];
  
  lastIndex(array2[1]);
  return firstIndex;
});

const output = myFunc();
console.log(output.result); // [0.08, 0.08, 0.08....
console.log(output[0]); // [2,2,2....
```

However, there is a new type detection capability in the v2 release candidate that I tested over lunch that resolves the function type without any difficulty, and I then began testing if I could get the output to match up to what we'd like, as:
```js
[
  [0.08, 2],
  [0.08, 2],
  [0.08, 2],
  [0.08, 2],
  [0.08, 2],
  ...
]
```

And while I'm not yet successful, I was able to get it to write to two separate textures, similarly to the above, but with full type detection.  I'll have to put this down for now, I have to work.  But I will revisit before v2's release, and at this time, I believe it will be part of it.

In the mean time, I'll continue this thread in the related issue I mentioned above.
